lib/if.c

```diff
@@ -668,21 +672,26 @@ int if_is_running(const struct interface *ifp)
    if ptm checking is enabled, then ptm check has passed */
 int if_is_operative(const struct interface *ifp)
 {
-	return ((ifp->flags & IFF_UP)
-		&& (((ifp->flags & IFF_RUNNING)
-		     && (ifp->ptm_status || !ifp->ptm_enable))
-		    || !CHECK_FLAG(ifp->status,
-				   ZEBRA_INTERFACE_LINKDETECTION)));
+	return (((ifp->flags & IFF_UP)
+#ifdef IFF_LOWER_UP
+		 && (ifp->flags & IFF_LOWER_UP)
```

Comment:
I don't think this is correct. `IFF_LOWER_UP` should be checked only if `ZEBRA_INTERFACE_LINKDETECTION` is enabled, this is the purpose of link-detection flag. You should move the check into the same parentheses with `IFF_RUNNING` check.

Reply:
done